weblate/api/tests.py

```diff
@@ -1235,6 +1237,54 @@ def test_patch(self):
         )
         self.assertEqual(response.data["slug"], "new-slug")
 
+    def test_create_component_docfile(self):
+        with open(TEST_DOC, "rb") as handle:
+            response = self.do_request(
+                "api:project-components",
+                self.project_kwargs,
+                method="post",
+                code=201,
+                superuser=True,
+                request={
+                    "docfile": handle,
+                    "name": "Local project",
+                    "slug": "local-project",
+                    "repo": "local:",
+                    "vcs": "local",
+                    "filemask": "*.strings",
+                    "template": "en.strings",
+                    "file_format": "strings-utf8",
```

Comment:
I'm surprised that this works - you're uploading plain text file and parsing it as strings, what might be a bug in the parser that it accepts it. 

```suggestion
                    "file_format": "html",
```